### PR TITLE
mysql_async, disable default features (#285)

### PR DESCRIPTION
**Pull request by glueball**
Disable default features for mysql_async. Enable only the minimal feature, which is required by the mysql_crate.

Also, removed the dependency with the tokio feature. Doesn't seem to be needed, since tokio isn't mentioned at all in the refinery_core::drivers::mysql_async module.

I was not sure how to test this, I did this and everything works:

```
test --workspace -F mysql_async -- --test-threads=1
```

### CODE REVIEW

**Comment by glueball** [comment]
Also, I think you could enable "minimal" in regular mysql instead of the manual flate2 dependency.

Some other dependencies could have some features removed also. Most notably tokio, is really "full" needed by refinery? Enabling features in refinery prevents downstream users from disabling them.

**Comment by jxs** [comment]
Hi, and thanks for this! :heart: 

>Also, I think you could enable "minimal" in regular mysql instead of the manual flate2 dependency.
>
>Some other dependencies could have some features removed also. Most notably tokio, is really "full" needed by refinery? >Enabling features in refinery prevents downstream users from disabling them.

Yeah, with time I have been made aware (and suffered because :sweat_smile: ) of that, do you wanna go ahead and do it on this PR? we can update the wording to reflect the changes, I can then release a tiny as this shouldn't be breaking.

**Review by jxs — APPROVED** [review]

**Comment by glueball** [comment]
Yeah I can add that to the pr!

**Comment by glueball** [comment]
However, maybe this can be considered breaking. Not sure what the actual policy for that is. 

Some downstream projects might be using features enabled now by refinery that are not listed in their own Cargo.toml. The fix is trivial, and if using rust 1.72 even the compiler warns what features to enable; but technically can be considered breaking.

**Comment by glueball** [comment]
Used [cargo hack]() to check all possible feature combinations

```
cargo hack check --feature-powerset --no-dev-deps --skip default --workspace
```

I also took the liberty to update siphasher to 1.0, as my IDE was giving me a warning. The hasher seems to be covered by migration divergence tests, so if CI turns green it should be safe.

File: refinery_core/Cargo.toml [hunk 1]
```diff
@@ -33,17 +33,14 @@ walkdir = "2.3.1"
...
 tokio-postgres = { version = ">= 0.5, <= 0.7", optional = true }
-mysql = { version = ">= 21.0.0, <= 23", optional = true, default-features = false }
-mysql_async = { version = ">= 0.28, <= 0.32", optional = true }
+mysql = { version = ">= 21.0.0, <= 23", optional = true, default-features = false, features = ["minimal"] }
+mysql_async = { version = ">= 0.28, <= 0.32", optional = true, default-features = false, features = ["minimal"] }
 tiberius = { version = ">= 0.7, <= 0.12", optional = true, default-features = false }
-tokio = { version = "1.0", features = ["full"], optional = true }
-futures = { version = "0.3.16", optional = true }
-tokio-util = { version = "0.7.7", features = ["compat"], optional = true }
+tokio = { version = "1.0", default-features = true, optional = true }
```
Comment (jxs) [inline refinery_core/Cargo.toml:39]:
```suggestion
tokio = { version = "1.0", optional = true }
```
default features is true by default

File: refinery_core/Cargo.toml [hunk 2]
```diff
@@ -33,17 +33,14 @@ walkdir = "2.3.1"
...
+futures = { version = "0.3.16", optional = true, default-features = true, features = ["async-await"] }
```
Comment (jxs) [inline refinery_core/Cargo.toml:40]:
```suggestion
futures = { version = "0.3.16", optional = true, features = ["async-await"] }
```

File: refinery_core/Cargo.toml [hunk 3]
```diff
@@ -33,17 +33,14 @@ walkdir = "2.3.1"
...
+tokio-util = { version = "0.7.7", features = ["compat"], default-features = true, optional = true }
```
Comment (jxs) [inline refinery_core/Cargo.toml:41]:
```suggestion
tokio-util = { version = "0.7.7", features = ["compat"], optional = true }
```

**Comment by jxs** [comment]
> However, maybe this can be considered breaking. Not sure what the actual policy for that is.
> 
> Some downstream projects might be using features enabled now by refinery that are not listed in their own Cargo.toml. The fix is trivial, and if using rust 1.72 even the compiler warns what features to enable; but technically can be considered breaking.

yeah thanks for mentioning, I can release a minor instead

**Review by jxs — APPROVED** [review]

**Comment by jxs** [comment]
thinking more about it, I am willing to run that risk. I hope there are not a lot of users doing it, but misusing the feature set should not be considered a feature I'd say, therefore we won't be breaking anything.
[released 0.8.11](https://github.com/rust-db/refinery/pull/287)

**Pull request merged**